### PR TITLE
docs: drop redundant defaults

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

these defaults are already shared as the footprint of function/method so having then in docs is duplicate and also not often updated

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: src/pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -195,111 +193,87 @@ def __init__(
                 trying to optimize initial learning for faster convergence. trainer.tune() method will
                 set the suggested learning rate in self.lr or self.learning_rate in the LightningModule.
                 To use a different key set a string instead of True with the key name.
-                Default: ``False``.
```
Comment (awaelchli) [inline src/pytorch_lightning/trainer/trainer.py:198]: Some of these were added by contributors: #11614

As a general guideline I agree, but it also doesn't hurt having them in something like the core Trainer API (that doesn't really change their defaults that often). 

I propose to keep these removals to a minimum, and to keep them in the Trainer.

Comment (Borda) [reply src/pytorch_lightning/trainer/trainer.py:198]: I see, not sure why we moved some back...
In my point of view, I am very fine to keep the default when there is some additional explanation or referring to the class type or just in text after a value set that the one is default... but having pure sentence "Default: None." I think is useless...

Comment (carmocca) [reply src/pytorch_lightning/trainer/trainer.py:198]: I suggest keeping the "default:" message for the arguments that are Optional in the code only so that we can know whether the user manually passed an argument or not to then set the real default. All others could be removed

Comment (Borda) [reply src/pytorch_lightning/trainer/trainer.py:198]: > arguments that are Optional in the code only so that we can know whether the user manually passed an argument

not sure if I understand what you mean here :confused:

Comment (carmocca) [reply src/pytorch_lightning/trainer/trainer.py:198]: We often use the pattern

```py
x: Optional[int]
if x is None:
    # the user didn't set it
    self.x = 10
else:
    self.x = x
```
instead of

```pyt
x: int = 10
self.x = x
```

So for those, we want to still mention `Default: 10` in the docstring because saying `Default: None` is not informative unless it also explains that `None` gets converted into `10`

Comment (Borda) [reply src/pytorch_lightning/trainer/trainer.py:198]: yes agree, but so far it was not such and we just said default None...

**Review by tchaton — APPROVED** [review]
LGTM 1

File: src/pytorch_lightning/trainer/trainer.py (hunk 1 above)
Comment (awaelchli) [reply src/pytorch_lightning/trainer/trainer.py:202]: I'm ok with adjusting the other less important files. But my opinion remains unchanged regarding trainer.py, I'd like that it stays and that we don't go back and forth #11614

Comment (awaelchli) [reply src/pytorch_lightning/trainer/trainer.py:202]: But @Borda how about we merge the PR but remove the changes from Trainer. That's what I'm trying to say.
